### PR TITLE
Add group asymmetry estimate to new GUI

### LINKED ISSUE #24646: Add Group range boxes to new Muon GUI
The new GUI needs an option to specify the time range to use when estimating group asymmetry. The group asymmetry should also be estimated for each group specified.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This PR adds the estimation of group asymmetry values to the new GUI. It also adds the ability to select the ranges used to estimate the group asymmetries to the grouping tab

**Report to:** anthony 

**To test:**

Open up the new GUI and load some data. Check that a group asymmetry estimate is calculated. Change the ranges on the grouping tab and check that the ranges used to estimate the group asymmetry are updated appropriately. Check that it also works for multiperiod and co-added data.

Fixes #24646  

*This does not require release notes* because the new GUI is not currently available to users

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/calculate_pair_and_group.py [hunk 1]
```diff
@@ -26,6 +26,15 @@ def calculate_pair_data(context, pair_name, run):
 
     return pair_data
 
+def estimate_group_asymmetry_data(context, group_name, run):
+    processed_data = _run_pre_processing(context, run)
+
+    params = _get_MuonGroupingAsymmetry_parameters(context, group_name, run)
+    params["InputWorkspace"] = processed_data
+    pair_data = algorithm_utils.run_MuonGroupingAsymmetry(params)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/calculate_pair_and_group.py]: I think we should change the variable name, as far as I can tell this is not a muon pair

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py [hunk 2]
```diff
@@ -140,4 +148,38 @@ def update_view_from_model(self):
...
+        else:
+            self._view.group_range_min.setEnabled(True)
+            self._model._data.gui_variables['GroupRangeMin'] = float(self._view.group_range_min.text())
+
+    def from_file_checkbox_changed(self):
+        if self._view.group_range_use_last_data.isChecked():
+            if self._model._data.current_runs:
+                self._view.group_range_max.setText(str(round(max(self._model._data.loaded_data(self._model._data.
+                                                                                               current_runs[-1])['OutputWorkspace'][0].
+                                                                 workspace.dataX(0)), 3)))
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py]: whats the significance of the 3?

**Comment by AnthonyLim23** [comment]
just tried it and the output is close enough to the original (difference at about 4 sf).

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py]: This is the number of decimal places to round to before displaying

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py]: could we make it a variable with a descriptive name?

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**